crates/viewer/re_viewer_context/src/component_fallbacks.rs

```diff
@@ -37,87 +48,241 @@ pub enum ComponentFallbackError {
     UnexpectedEmptyFallback,
 }
 
-/// Provides fallback values for components, implemented typically by [`crate::ViewClass`] and [`crate::VisualizerSystem`].
+type ComponentFallbackProviderFn =
+    Box<dyn Fn(&QueryContext<'_>) -> Result<ArrayRef, SerializationError> + Send + Sync + 'static>;
+
+/// A registry to handle component fallbacks.
 ///
-/// Fallbacks can be based on arbitrarily complex & context sensitive heuristics.
-pub trait ComponentFallbackProvider {
-    /// Tries to provide a fallback value for a given component.
-    ///
-    /// If the provider can't handle the component or simply want to use a placeholder value,
-    /// it should return [`ComponentFallbackProviderResult::ComponentNotHandled`].
+/// This has 5 layers of fallbacks:
+/// - First try to use a fallback for the view context and [`ComponentIdentifier`].
```

Comment:
changed a bit / can be more accurate:

```suggestion
/// - First try to use a fallback using the active type of view (identified via [`ViewClassIdentifier`]) and [`ComponentIdentifier`].
```